Assets/Scripts/Framework/PhoneCamera/PhotoInteractable.cs

```diff
@@ -1,22 +1,94 @@
-﻿using UnityEngine;
+﻿using System;
+using System.IO;
+using UnityEngine;
 
 using FrameWork;
 using Framework.ScriptableObjects;
-using UI.Canvas.PhoneCamera;
+using UI.Canvas.PhoneBook;
 
 namespace Framework.PhoneCamera
 {
     public sealed class PhotoInteractable : InteractableObject
     {
+        private const string NO_TEXTURE_TO_SAVE = "No texture to save.";
+        private const string NO_TEXTURE_TO_LOAD = "No texture data to load.";
+        private const string SAVED_TEXTURE = "savedTexture.png";
+        
         [SerializeField] private PhotoTaker parent;
-        [SerializeField] private PhotoData data;
+        [SerializeReference] private PhotoData data;
+        
+        private byte[] _textureBytes;
         
-        public PhotoData Data => data;
+        public bool IsVisited { get; set; }
+        
+        public Page ParentPage { get; set; }
 
+        /// <summary>
+        /// Set this PhotoInteractable as the current in the PhotoTaker
+        /// </summary>
         public void SetPhotoData()
         {
+            parent.SetCurrentPhotoInteractable(this);
+            
             if(parent.Data != data)
                 parent.Data = data;
         }
+
+        /// <summary>
+        /// Get the text to show in the page, if visited will show the text for it, otherwise the default text
+        /// </summary>
+        /// <returns></returns>
```

Comment:
```suggestion
        /// <returns> Returns text for text block depending if the photo has been taken or not </returns>
```
Staat niets in comment :(